api/src/main/java/org/pmiops/workbench/api/UserController.java

```diff
@@ -114,43 +116,125 @@ public UserController(
       throw new ForbiddenException("user search requires registered data access");
     }
 
-    Sort.Direction direction =
-        Sort.Direction.fromOptionalString(sortOrder).orElse(Sort.Direction.ASC);
-    Sort sort = Sort.by(new Sort.Order(direction, DEFAULT_SORT_FIELD));
+    // What we are really looking for here are users who have a FC account.
+    // This should exist if they have signed in at least once
+    List<DbUser> users =
+        userService.findUsersBySearchString(term, getSort(sortOrder)).stream()
+            .filter(user -> user.getFirstSignInTime() != null)
+            .collect(Collectors.toList());
+
+    return processSearchResults(term, pageSize, response, paginationToken, users);
+  }
+
+  /**
+   * Return a page of users matching a search term and an access tier. Used by autocomplete for
+   * workspace sharing.
+   *
+   * @param accessTierShortName the shortName of the access tier to search in; the calling user must
+   *     also be a member of this tier
+   * @param term a search term to match against the user's name and username fields (case
+   *     insensitive)
+   * @param pageToken Pagination token retrieved from a previous call to user; used for retrieving
+   *     additional pages of results.
+   * @param pageSize Maximum number of results to return in a response. Defaults to 10.
+   * @param sortOrder 'asc' or 'desc', defaulting to 'asc'
+   * @return A list of users matching the provided search query and a nextPageToken if applicable.
+   */
+  @Override
+  public ResponseEntity<UserResponse> userSearch(
+      String accessTierShortName,
+      String term,
+      String pageToken,
+      Integer pageSize,
+      String sortOrder) {
+    UserResponse response = initializeUserResponse();
+
+    if (null == term || term.isEmpty()) {
+      return ResponseEntity.ok(response);
+    }
+
+    PaginationToken paginationToken;
+    try {
+      paginationToken = getPaginationTokenFromPageToken(pageToken);
+    } catch (IllegalArgumentException | BadRequestException e) {
+      return ResponseEntity.badRequest().body(response);
```

Comment:
Do we need to log/put exception e into response or in the log?

Reply:
logged.

UserResponse does not have a way to show an error message.